### PR TITLE
chore: Add abstraction layer around ACIR gen of quotient directives

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

## Problem\*

Resolves 

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by kevaundray** [comment]
The bigger issue here is the fact that we are liberally creating witnesses out of thin air. An abstraction over this like `range_directive` would help here

**Comment by TomAFrench** [comment]
> An abstraction over this like range_directive would help here

I can add this in this PR if you're not working on this already.

**Comment by kevaundray** [comment]
> > An abstraction over this like range_directive would help here
> 
> I can add this in this PR if you're not working on this already.

Sorry forgot to reply -- yeah this would be good, I have a few things in the backlog, so it would be quicker if you did it

**Comment by TomAFrench** [comment]
Easiest to do it here. Reordering the opcodes is only a substantive change in the context of #1729 so we can just make this PR be primarily about the new abstraction layer.

**Comment by TomAFrench** [comment]
@kevaundray if you give an ok in the comments here, I can approve + merge.

**Comment by TomAFrench** [comment]
Looks like we should add the range constraints into the helper as well.

**Comment by kevaundray** [comment]
LGTM

**Review by TomAFrench — APPROVED** [review]

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs [hunk 1]
```diff
@@ -680,6 +664,30 @@ impl GeneratedAcir {
         Ok(())
     }
 
+    /// Adds a directive which injects witnesses with values `q = a / b` and `r = a % b`.
```
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs:667]: nit: We probably want to note that we add constraints to ensure that quotient and remainder are range constrained

**Review by TomAFrench — APPROVED** [review]

**Pull request merged**